polaris.shopify.com/src/components/Markdown/components/DoDont/index.tsx

```diff
@@ -44,8 +43,7 @@ export const DoDont = ({children}: PropsWithChildren) => {
   while (i < childrenArray.length) {
     target = childrenArray[i].props.children.startsWith('Don') ? donts : dos;
 
-    // Add the heading
-    target.push(childrenArray[i]);
+    // skip the headings in older uses of <DoDont>
```

Comment:
removed the heading since it looks redundant with the new design

Before
<img width="2032" alt="Screenshot 2023-09-20 at 11 19 31 AM" src="https://github.com/Shopify/polaris/assets/4642404/749c0f43-1a59-4436-acc2-c6e2ec691a68">

After 
<img width="2032" alt="Screenshot 2023-09-20 at 11 15 08 AM" src="https://github.com/Shopify/polaris/assets/4642404/496334e0-a7fd-4a94-a68c-f10bc4c86356">